File: archive/c/c-sharp/Capitalize.cs
if (args is not [string input, ..] || string.IsNullOrWhiteSpace(input))
{
    Console.WriteLine("Usage: please provide a string");
    return;
}

if (char.IsUpper(input[0]))
{
    Console.WriteLine(input);
    return;
}

string output = string.Create(
    input.Length,
    input,
    static (span, str) =>
    {
        str.CopyTo(span);
        span[0] = char.ToUpperInvariant(span[0]);
    }
);

Console.WriteLine(output);

File: archive/v/visual-basic/capitalize.vb
Module Capitalize

    Sub Main(args As String())

        If args.Length = 0 OrElse String.IsNullOrWhiteSpace(args(0)) Then
            Console.WriteLine("Usage: please provide a string")
            Return
        End If

        Dim input = args(0)

        Dim output =
            Char.ToUpperInvariant(input(0)) &
            If(input.Length > 1, input.Substring(1), String.Empty)

        Console.WriteLine(output)

    End Sub

End Module